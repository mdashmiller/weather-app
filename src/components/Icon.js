import styled from "styled-components"

const Icon = styled.i`
    display: inline-block;
    width: 40px;
    margin: 10px;
    color: #c3c3c3;
`

export default Icon
