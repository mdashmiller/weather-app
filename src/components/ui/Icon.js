import styled, { css } from "styled-components"

const Icon = styled.i`
    display: inline-block;
    width: 40px;
    margin: 20px;
    color: #c3c3c3;

    ${props => props.dark && css`
        color: #818181;
      `}
`

export default Icon
