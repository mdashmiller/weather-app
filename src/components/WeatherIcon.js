import React from 'react'
import Icon from './Icon'

const WeatherIcon = ({ id, daytime }) => {
	// determines whether to display a
	// cloud or sun icon dependent upon
	// weather conditions

	// weather-dependent icon details
	let celestialBody
	if (daytime) {
		celestialBody = "fas fa-sun fa-3x"
	} else {
		celestialBody = "fas fa-moon fa-3x"
	}
	const cloud = "fas fa-cloud fa-3x"	

	if (
			(id > 959 && id < 963) ||
			(id > 899 && id < 903) ||
			id === 804 ||
			id === 803 ||
			(id > 700 && id < 782) ||
			(id > 599 && id < 623) ||
			(id > 499 && id < 532) ||
			(id > 299 && id < 322) ||
			(id > 199 && id < 233)
	) {
		return <Icon dark className={cloud}></Icon>
	}
	return <Icon className={celestialBody}></Icon>
}
	
export default WeatherIcon
