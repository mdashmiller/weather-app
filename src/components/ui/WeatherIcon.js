import React from 'react'
import Icon from './Icon'
import PropTypes from 'prop-types'

const WeatherIcon = ({ code, day }) => {
	// determines whether to display a
	// cloud or sun icon dependent upon
	// weather conditions

	// weather-dependent icon details
	const celestialBody = day
		? "fas fa-sun fa-3x" : "fas fa-moon fa-3x"
	const cloud = "fas fa-cloud fa-3x"	

	if (
			(code > 959 && code < 963) ||
			(code > 899 && code < 903) ||
			code === 804 ||
			code === 803 ||
			(code > 700 && code < 782) ||
			(code > 599 && code < 623) ||
			(code > 499 && code < 532) ||
			(code > 299 && code < 322) ||
			(code > 199 && code < 233)
	) {
		return <Icon dark className={cloud}></Icon>
	}
	return <Icon className={celestialBody}></Icon>
}

WeatherIcon.propTypes = {
	code: PropTypes.number,
	day: PropTypes.bool
}

export default WeatherIcon
