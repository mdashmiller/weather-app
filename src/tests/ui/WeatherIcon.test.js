import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import WeatherIcon from '../../components/ui/WeatherIcon'
import Icon from '../../components/ui/Icon'

Enzyme.configure({ adapter: new Adapter })

describe('<WeatherIcon /> rendering', () => {

	it('renders without crashing', () => {
	  	const div = document.createElement('div')
	  	ReactDOM.render(<WeatherIcon />, div)
	  	ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(<WeatherIcon />)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

})

describe('<WeatherIcon /> displays proper className for each condition code', () => {

	it('has className {cloud} when it receives a cloudy weather code', () => {
		let wrapper = shallow(<WeatherIcon code={960} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={900} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={804} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={803} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={760} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={600} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={500} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={310} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
		wrapper = shallow(<WeatherIcon code={210} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-cloud fa-3x')
	})

	it('displays the proper celestial body when it receives a clear weather code', () => {
		let wrapper = shallow(<WeatherIcon code={199} day={true} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-sun fa-3x')
		wrapper = shallow(<WeatherIcon code={199} day={false} />)
		expect(wrapper.find(Icon).prop('className')).toBe('fas fa-moon fa-3x')
	})

})
