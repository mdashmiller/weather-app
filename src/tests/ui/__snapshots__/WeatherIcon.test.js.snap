// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherIcon /> rendering has a valid snapshot 1`] = `
<i
  className="fas fa-moon fa-3x sc-bdVaJa bSIgLK"
/>
`;
